src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void add(T value) {
-
+        if (objects.length == size) {
+            increaseObjectsLength();
+        }
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        validateIndex(index);
+        Object[] increasedObjects = new Object[objects.length + 1];
+        if (index > 0) {
+            System.arraycopy(objects, 0, increasedObjects, 0, index);
+        }
+        System.arraycopy(objects, index, increasedObjects, index + 1, objects.length - index);
+        increasedObjects[index] = value;
+        objects = increasedObjects;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        validateIndex(index);
+        return (T) objects[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        validateIndex(index);
+        objects[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        validateIndex(index);
+        Object[] decreasedObjects = new Object[objects.length - 1];
```

Comment:
my bad, i confused you on QnA. Actual java.util.Arraylist doesn't trim itself after you remove object from it, its expensive for performance. That's just for you info.

Reply:
Ok, thank you. I will keep it in mind.